components/toast_wrapper/toast_wrapper.tsx

```diff
@@ -2,103 +2,95 @@
+
+import type {OwnProps, PropsFromRedux} from '.';
 
 const TOAST_TEXT_COLLAPSE_WIDTH = 500;
 
 const TOAST_REL_RANGES = [
     RelativeRanges.TODAY_YESTERDAY,
 ];
 
-class ToastWrapper extends React.PureComponent {
-    static propTypes = {
-        unreadCountInChannel: PropTypes.number,
-        newRecentMessagesCount: PropTypes.number,
-        channelMarkedAsUnread: PropTypes.bool,
-        isCollapsedThreadsEnabled: PropTypes.bool,
-        rootPosts: PropTypes.object,
-        atLatestPost: PropTypes.bool,
-        postListIds: PropTypes.array,
-        latestPostTimeStamp: PropTypes.number,
-        atBottom: PropTypes.bool,
-        lastViewedBottom: PropTypes.number,
-        width: PropTypes.number,
-        lastViewedAt: PropTypes.number,
-        focusedPostId: PropTypes.string,
-        initScrollOffsetFromBottom: PropTypes.number,
-        updateNewMessagesAtInChannel: PropTypes.func,
-        scrollToNewMessage: PropTypes.func,
-        scrollToLatestMessages: PropTypes.func,
-        scrollToUnreadMessages: PropTypes.func,
-        updateLastViewedBottomAt: PropTypes.func,
-        showSearchHintToast: PropTypes.bool,
-        onSearchHintDismiss: PropTypes.func,
-        shouldStartFromBottomWhenUnread: PropTypes.bool,
-        isNewMessageLineReached: PropTypes.bool,
-        unreadScrollPosition: PropTypes.string,
-
-        /*
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            params: PropTypes.shape({
-                team: PropTypes.string,
-            }).isRequired,
-        }).isRequired,
-
-        actions: PropTypes.shape({
-
-            /**
-             * Action creator to update toast status
-             */
-            updateToastStatus: PropTypes.func.isRequired,
-        }).isRequired,
-    };
-
+export type Props = PropsFromRedux & OwnProps & WrappedComponentProps & RouteComponentProps<{team: string}> & {
+    channelMarkedAsUnread?: boolean;
+    postListIds: string[];
+    latestPostTimeStamp?: number;
+    atBottom: boolean | null;
+    lastViewedBottom: number;
+    width: number;
+    focusedPostId?: string;
+    initScrollOffsetFromBottom: number;
+    updateNewMessagesAtInChannel: (lastViewedAt?: number) => void;
+    scrollToNewMessage: () => void;
+    scrollToLatestMessages: () => void;
+    scrollToUnreadMessages: () => void;
+    updateLastViewedBottomAt: (lastViewedBottom?: number) => void;
+    showSearchHintToast: boolean;
+    onSearchHintDismiss: () => void;
+    shouldStartFromBottomWhenUnread: boolean;
+    isNewMessageLineReached: boolean;
+};
+
+type State = {
+    unreadCount?: number;
+    unreadCountInChannel: number;
+    channelMarkedAsUnread?: boolean;
+    lastViewedAt?: number;
+    showUnreadToast?: boolean;
+    showNewMessagesToast?: boolean;
+    showMessageHistoryToast?: boolean;
+    showUnreadWithBottomStartToast?: boolean;
+};
+
+export class ToastWrapperClass extends React.PureComponent<Props, State> {
```

Comment:
Do we need the name change here?

Reply:
Yes, this was needed because of tests. I am using `ToastWrapperClass` to correctly get the types such as `(wrapper.instance() as ToastWrapperClass).scrollToLatestMessages();` 

What I usually do with shallow to overcome this issue: `shallow<ToastWrapper>` but here `shallowWithIntl` is used, passing a generic was not possible. Please let me know if there is any other way to fix this issue.